microgen/rve.py

```diff
@@ -22,45 +22,52 @@ class Rve:
     :param dim: dimensions of the RVE
     """
 
-    def __init__(
+    def __init__(  # pylint: disable=too-many-arguments
         self,
         dim_x: Optional[float] = None,
         dim_y: Optional[float] = None,
         dim_z: Optional[float] = None,
         center: Union[np.ndarray, Tuple, List] = (0, 0, 0),
-        dim: Optional[Union[float, np.ndarray, Tuple, List]] = 1,
+        dim: Union[float, np.ndarray, Tuple, List] = 1,
     ) -> None:
-        if isinstance(center, (Tuple, List)) and len(center) == _DIM:
+        if isinstance(center, (tuple, list)) and len(center) == _DIM:
             self.center = np.array(center)
         elif isinstance(center, np.ndarray) and center.shape == (_DIM,):
             self.center = center
         else:
             raise ValueError(f"center must be an array or Sequence of length {_DIM}")
 
-        if (
-            dim is not None
-            and dim_x is not None
-            and dim_y is not None
-            and dim_z is not None
-        ):
-            raise ValueError("Either dim or dim_x, dim_y, dim_z must be specified")
-
         if isinstance(dim, (int, float)):
             self.dim = np.array([dim for _ in range(_DIM)])
-        elif isinstance(dim, (Tuple, List)) and len(dim) == _DIM:
+        elif isinstance(dim, (tuple, list)) and len(dim) == _DIM:
             self.dim = np.array(dim)
         elif isinstance(dim, np.ndarray) and dim.shape == (_DIM,):
             self.dim = dim
-        elif dim is None:
-            if dim_x is None or dim_y is None or dim_z is None:
-                raise ValueError("dim must be specified")
+        else:
+            raise ValueError(f"dim must be an array or Sequence of length {_DIM}")
+
+        if dim_x is not None and dim_y is not None and dim_z is not None:
+            self.dim = np.array([dim_x, dim_y, dim_z])
             warnings.warn(
-                "dim_x, dim_y, dim_z will be deprecated, use 'dim' instead.",
+                f"dim_x, dim_y, dim_z are deprecated, use 'dim' instead. \
+                    Now dim is set to [{self.dim[0]}, {self.dim[1]}, {self.dim[2]}]",
                 DeprecationWarning,
+                stacklevel=2,
+            )
+        elif dim_x is not None or dim_y is not None or dim_z is not None:
```

Comment:
In which case we are in this condition ?

Reply:
In case of `Rve(dim_x=2)` to get `dim == [2, 1, 1]` as it would be in microgen 1.0.0

Reply:
Refactored this part to take care of both situations in the same if